pkg/storage/tsdb/index_cache.go

```diff
@@ -60,17 +60,22 @@ func (cfg *IndexCacheConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix str
 
 // Validate the config.
 func (cfg *IndexCacheConfig) Validate() error {
-	if !util.StringsContain(supportedIndexCacheBackends, cfg.Backend) {
-		return errUnsupportedIndexCacheBackend
-	}
 
-	if cfg.Backend == IndexCacheBackendMemcached {
-		if err := cfg.Memcached.Validate(); err != nil {
-			return err
+	splitedBackends := strings.Split(cfg.Backend, ",")
```

Comment:
Should we check that the backends are unique here in the validation?

So that the users won't put "inmemory,inmemory,memcached,redis"
And also limit the number of levels?

Reply:
Humm...

I think someone have 2 memcache caches no? one closer - on the same az - and one cross Az? Idk if we should limit this.

WDYT?

Reply:
I don't think the current config allows the user to specify multiple instances for the same backend. The backends are YAML map entries.

```
index_cache:
    [backend: <string> | default = "inmemory"]

    inmemory:
      [max_size_bytes: <int> | default = 1073741824]

    memcached:
      [addresses: <string> | default = ""]

    redis:
      [addresses: <string> | default = ""]
```

A validation will be good to have.

Reply:
Gotcha! make sense